cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/DFGPass.kt

```diff
@@ -398,11 +404,13 @@ class DFGPass(ctx: TranslationContext) : ComponentPass(ctx) {
             handleUnresolvedCalls(call, call)
         } else if (call.invokes.isNotEmpty()) {
             call.invokes.forEach {
-                if (it.isInferred && inferDfgForUnresolvedSymbols) {
-                    handleUnresolvedCalls(call, it)
-                } else {
-                    Util.attachCallParameters(it, call.arguments)
-                    call.addPrevDFG(it)
+                Util.attachCallParameters(it, call.arguments)
+                if (it is MethodDeclaration && call is MemberCallExpression && !call.isStatic) {
+                    it.receiver?.let { receiver -> call.base?.addNextDFG(receiver) }
+                }
+                call.addPrevDFG(it)
+                if (it.isInferred && it.prevDFG.isEmpty()) {
+                    handleFunctionDeclaration(it)
```

Comment:
We should add a comment here, why we are doing it this way, i.e., that inferred nodes are most likely not in the AST (yet) and that handleFunctionDeclaration only iterates on AST functions.

Reply:
I added a comment for it.